crates/bgp-pkt/src/path_attribute/bgp_sid.rs

```diff
@@ -0,0 +1,131 @@
+use crate::{
+    iana::BgpSidAttributeType, nlri::MplsLabel, path_attribute::PathAttributeValueProperties,
+};
+use serde::{Deserialize, Serialize};
+
+/// 3 bytes MPLS Label
+/// followed by
+/// 3 bytes for range size
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[cfg_attr(feature = "fuzz", derive(arbitrary::Arbitrary))]
+pub struct SRGB {
+    pub first_label: MplsLabel,
+    pub range_size: [u8; 3],
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[cfg_attr(feature = "fuzz", derive(arbitrary::Arbitrary))]
+pub enum BgpSidAttribute {
+    /// ```text
+    ///  0                   1                   2                   3
+    ///  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+    /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /// |       Type    |             Length            |   RESERVED    |
+    /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /// |            Flags              |       Label Index             |
+    /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    /// |          Label Index          |
+    /// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    ///
+    /// where:
+    ///
+    /// Type: 1
+    /// Length: 7, the total length in octets of the value portion of the TLV.
+    /// RESERVED: 8-bit field. It MUST be clear on transmission and MUST be ignored on reception.
+    /// Flags: 16 bits of flags. None are defined by this document.
+    /// The Flags field MUST be clear on transmission and MUST be ignored on reception.
+    /// Label Index: 32-bit value representing the index value in the SRGB space.
+    /// ```
+    LabelIndex {
+        /// None defined in RFC8669
```

Comment:
what does None defined in RFC8669 means?

Reply:
I would put that as a comment then ;-)

Reply:
i think it's nice as a documentation because if you hover on the field while creating the struct you'll get the tip that this should be 0

i can expand the doc a bit though, something like the following would be more complete
```
Flags for the SR Label Index are not yet defined (RFC8669). 
Therefore, this field MUST be left blank (0) for now.
```

what do you think?

Reply:
yeah that works, thank you